Fix JUri::root() assignment from backend app constuctor

izharaazmi opened the pull request:
This may not seem any important but I'd say it still is breaking. Try running Joomla in a subdirectory `administrator` under your web root, and open backend. It would break.

```
example.com/administrator/administrator/index.php
```

`dirname` will work on all platforms as `JUri::base(true)` always uses `/` as the path separator. Also there is no trailing slash in the return value of  `JUri::base(true)`

izharaazmi committed:
Fix JUri::root() assignment from backend app constuctor

This may not seem any important but I'd say it still is breaking. Try running Joomla in a subdirectory `administrator` under your web root, and open backend. It would break.

brianteeman commented:
I have tested this item :white_check_mark: successfully on fb59736bb5959b8f4a7d27ea6b80bfee1d68a188<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8587">issues.joomla.org/joomla-cms/8587</a>.</sub>

izharaazmi committed:
dirname gives a slash for 1st level subdirectory 

Removed trailing slash, which eventually only occurs when queried for a root domain Joomla setup, i.e. not in a subdirectory installation.

pe7er commented:
Patch tested successfully on Joomla! 3.6.0-beta2

I've installed Joomla in a folder called /administrator/ and going to the /administrator/administrator/ results in a blank page.

I've manually added the code of this patch to the file, and it solves the problem.
Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8587">issues.joomla.org/joomla-cms/8587</a>.</sub>

pe7er commented:
I have tested this item :white_check_mark: successfully on 925000824d46e42f225da35f97a68eb52c2e5761<br /><br />Patch tested successfully on Joomla! 3.6.0-beta2

I've installed Joomla in a folder called /administrator/ and going to the /administrator/administrator/ results in a blank page.

I've manually added the code of this patch to the file, and it solves the problem.

Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8587">issues.joomla.org/joomla-cms/8587</a>.</sub>

BezouwenR commented:
I have tested this item :white_check_mark: successfully on 925000824d46e42f225da35f97a68eb52c2e5761<br /><br />Patch tested successfully on Joomla! 3.6.0-beta2

I've installed Joomla in a folder called /administrator/ and going to the /administrator/administrator/ results in a blank page.

I've manually added the code of this patch to the file, and it solves the problem.

// JUri::root(null, str_ireplace('/' . $this->getName(), '', JUri::base(true)));
   JUri::root(null, rtrim(dirname(JUri::base(true)), '/'));

Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8587">issues.joomla.org/joomla-cms/8587</a>.</sub>

ggppdk commented:
I have tested this item :white_check_mark: successfully on 925000824d46e42f225da35f97a68eb52c2e5761<br /><br />Also works for placing joomla in sub-folder like
/administrator*/

e.g.
/administrator_2/<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8587">issues.joomla.org/joomla-cms/8587</a>.</sub>

zero-24 commented:
RTC based on testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8587">issues.joomla.org/joomla-cms/8587</a>.</sub>

roland-d commented:
Thanks everybody
